daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf_dev/daml_lf_1.proto

```diff
@@ -1309,6 +1309,7 @@ message DefDataType {
     Fields record = 3; // Records without fields are explicitly allowed.
```

Comment:
Given that we always require type synonyms to be fully applied it might be worth considering to have a separate case in `Type` for a fully applied type synonym. That might make it a bit easier to expand the type synonym in various places.

Reply:
I prefer not to not have a special case in the encoding. 
We already have the message `Type.Con` which encodes applied type in compact way.